cmd/activator/main.go

```diff
@@ -132,6 +149,73 @@ func main() {
 		logger.Fatalw("Failed to create stats reporter", zap.Error(err))
 	}
 
+	// set up signals so we handle the first shutdown signal gracefully
+	stopCh := signals.SetupSignalHandler()
+
+	kubeInformerFactory := kubeinformers.NewSharedInformerFactory(kubeClient, defaultResyncInterval)
+	servingInformerFactory := servinginformers.NewSharedInformerFactory(servingClient, defaultResyncInterval)
+	endpointInformer := kubeInformerFactory.Core().V1().Endpoints()
+	revisionInformer := servingInformerFactory.Serving().V1alpha1().Revisions()
+
+	go endpointInformer.Informer().Run(stopCh)
+	go revisionInformer.Informer().Run(stopCh)
+
+	params := queue.BreakerParams{QueueDepth: breakerQueueDepth, MaxConcurrency: breakerMaxConcurrency, InitialCapacity: 0}
+
+	// Return the number of endpoints, 0 if no endpoints are found
+	endpointsGetter := func(revID activator.RevisionID) (int32, error) {
+		endpointKey := cache.ExplicitKey(revID.Namespace + "/" + reconciler.GetServingK8SServiceNameForObj(revID.Name))
+		endpoints, exists, err := endpointInformer.Informer().GetIndexer().Get(endpointKey)
+		if err != nil {
+			return 0, fmt.Errorf("Error getting endpoints for revision %v", revID)
+		}
+		if !exists {
+			// no endpoints exist yet
+			return 0, nil
+		}
+		if exists && endpoints != nil {
+			endpoint := endpoints.(*corev1.Endpoints).Subsets
+			addresses := activator.GetEndpointAddresses(endpoint)
+			return int32(addresses), nil
+		}
+		return 0, nil
+	}
+
+	revisionGetter := func(revID activator.RevisionID) (*v1alpha12.Revision, bool, error) {
+		revisionKey := cache.ExplicitKey(revID.Namespace + "/" + revID.Name)
+		rev, exists, err := revisionInformer.Informer().GetIndexer().Get(revisionKey)
+		if exists && rev != nil {
+			return rev.(*v1alpha12.Revision), exists, err
+		}
+		return nil, exists, err
+	}
+
+	throttlerParams := activator.ThrottlerParams{BreakerParams: params, Logger: logger, GetEndpoints: endpointsGetter, GetRevision: revisionGetter}
+	throttler := activator.NewThrottler(throttlerParams)
+
+	endpointInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		UpdateFunc: activator.UpdateEndpoints(throttler),
+	})
+
+	revisionInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
+		DeleteFunc: activator.DeleteBreaker(throttler),
+	})
+
+	kubeInformerFactory.Start(stopCh)
+	servingInformerFactory.Start(stopCh)
+
+	logger.Info("Waiting for informer caches to sync")
+
+	informerSyncs := []cache.InformerSynced{
+		endpointInformer.Informer().HasSynced,
+		revisionInformer.Informer().HasSynced,
+	}
+	for i, synced := range informerSyncs {
+		if ok := cache.WaitForCacheSync(stopCh, synced); !ok {
+			logger.Fatalf("failed to wait for cache at index %d to sync", i)
```

Comment:
do you care about closing `stopCh` for a cleaner termination just in case?

Reply:
I'm not sure if we really need to close it here - since we end up with `Fatalf` which will call `os.Exit` - nothing dangling should be left